scimma/client/subscribe.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python
+
+__author__ = "Bryce Cousins (bfc5288@psu.edu)"
+__description__ = "tools to receive and parse GCN circulars"
+
+
+import argparse
+
+import json
+
+from .io import Stream
+
+
+def print_gcn(gcn_dict, json_dump=False):
+    """Parse a gcn dictionary and print to stdout.
+
+    Args:
+      gcn_dict:  the dictionary object containing a GCN-formatted message
+
+    Returns:
+      None
+
+    """
+    if json_dump:
+        print(json.dumps(gcn_dict))
+    else:
+        gcn_headers = gcn_dict["header"]
+        gcn_body = gcn_dict["body"]
+
+        for key in gcn_headers:
+            line = key.upper() + ":\t" + gcn_headers[key]
+            print(line)
+
+        for line in gcn_body.splitlines():
+            print(line)
+
+
+# ------------------------------------------------
+# -- main
+
+
+def _add_parser_args(parser):
+    parser.add_argument(
+        "-b",
+        "--broker-url",
+        required=True,
+        help="Sets the broker URL (kafka://host[:port]/topic) from which to receive GCNs.",
+    )
```

Comment:
I know I caused this API incompatibility (my bad), but would it be possible to raise an issue about the inconsistency so we can address it later? Alternatively, we can change it in this PR depending on if you're willing:

```
    parser.add_argument(
        "url",
        metavar="URL",
        help="Sets the URL (kafka://host[:port]/topic) to publish GCNs to.",
    )
```

and then change the arguments appropriately in this module and in the tests.

Reply:
Thanks, I was planning to add this update after completing the initial PR to be consistent with the #54 changes. See new commits on the PR for the changes.